scipy/stats/_sampling.py

```diff
@@ -0,0 +1,1213 @@
+        "center": 0.0,
+    },
+    "rayleigh": {
+        "pdf": lambda x: x * math.exp(-0.5 * (x * x)),
+        "center": 1.0,
+    },
+    "semicircular": {
+        "pdf": lambda x: math.sqrt(1.0 - (x * x)),
+        "center": 0,
+    },
+    "wald": {
+        "pdf": wald_pdf,
+        "center": 1.0,
+    },
+    "weibull_max": {
+        "pdf": weibull_max_pdf,
+        "check_pinv_params": lambda a: 0.25 <= a <= 512,
+        "center": -1.0,
+    },
+    "weibull_min": {
+        "pdf": weibull_min_pdf,
+        "check_pinv_params": lambda a: 0.25 <= a <= 512,
+        "center": 1.0,
+    },
+}
+
+
+def _validate_qmc_input(qmc_engine, d, seed):
+    # Input validation for `qmc_engine` and `d`
+    # Error messages for invalid `d` are raised by QMCEngine
+    # we could probably use a stats.qmc.check_qrandom_state
+    if isinstance(qmc_engine, QMCEngine):
+        if d is not None and qmc_engine.d != d:
+            message = "`d` must be consistent with dimension of `qmc_engine`."
+            raise ValueError(message)
+        d = qmc_engine.d if d is None else d
+    elif qmc_engine is None:
+        d = 1 if d is None else d
+        qmc_engine = Halton(d, seed=seed)
+    else:
+        message = (
+            "`qmc_engine` must be an instance of "
+            "`scipy.stats.qmc.QMCEngine` or `None`."
+        )
+        raise ValueError(message)
+
+    return qmc_engine, d
+
+
+class CustomDistPINV:
+    def __init__(self, pdf, args):
+        self._pdf = lambda x: pdf(x, *args)
+
+    def pdf(self, x):
+        return self._pdf(x)
+
+
+class FastGeneratorInversion:
+    """
+    Fast sampling by numerical inversion of the CDF for a large class of
+    continuous distributions in `scipy.stats`.
+
+    Parameters
+    ----------
+    distname : str
+        Name of the distribution in `scipy.stats`. The list of supported
+        distributions can be found in the Notes section.
+    args : tuple of floats or empty tuple, optional
+        The shape parameters (without `loc` and `scale`) of the distribution.
+        Important note: Only a single set of shape parameters is allowed, e.g.,
+        for the Gamma distribution with shape parameter `p`, `p` has to be a
+        float, and for the beta distribution with shape parameters (a, b), both
+        a and b have to be floats. The default is (), i.e., the distribution
+        has no shape parameters.
+    domain : tuple of floats, optional
+        If one wishes to sample from a truncated/conditional distribution,
+        the domain has to be specified.
+        The default is None. In that case, the random variates are not
+        truncated, and the domain is inferred from the support of the
+        distribution.
+    loc : float, optional
+        The location parameter. Note that only floats and no arrays are
+        allowed. The default is 0.
+    scale : float, optional
+        The scale parameter. Note that an array of floatsis not allowed.
+        The default is 1.
```

Comment:
I thought it is cleaner and more explicit with a keyword, plus it is in line with the `rv_continuous` methods that separate the shape parameters and loc/scale

not having loc and scale would be very problematic since it creates an inconsistency to how parameters are defined in SciPy, e.g., you need `scale` to define the exponential distribution with shape `scale`